### PR TITLE
Fixed wrong thread url copied to clipboard

### PR DESCRIPTION
**Pull request by kurtisassad**
## Link to Issue
Closes: #3644

## Description of Changes
- Fix the Copy URL button for discussions

## Test Plan
- CA (click around) tested on local
- Clicked copy URL button on discussions page for first dydx thread - verified correct
- Clicked copy URL button on discussion page for first dydx thread - verified correct
- Clicked copy URL button on first and third comment for dydx thread with multiple comments - verified correct

### CODE REVIEW

**Review by jnaviask — CHANGES_REQUESTED** [review]
Overview page still broken -- otherwise, looks good!!

**Review by jnaviask — CHANGES_REQUESTED** [review]
@kurtisassad missing chain scope on overview page. Just shows localhost/discussion/12345 (no chain).

**Comment by CowMuon** [comment]
Also just to note PR title here needs to be improved (FFR). The title here makes it sound like a simple copy change (such as we do get from BD) but which it is not.

Better here would  be something closer to what Malik put for the issue, "osmosis/proposals."

In any case,  this is not quite working yet in testing so converting  back to draft.

**Comment by jnaviask** [comment]
Also still broken on topics listing pages -- copies of topic at end of link.

Please rethink how you're constructing links here. We shouldn't be replacing the various page URLs. We should instead construct the link directly using the origin, thread id, and chain id (unless on custom domain). The code changes in this PR are extremely hard to follow as a result of the unclear logic.

**Comment by kurtisassad** [comment]
> Also still broken on topics listing pages -- copies of topic at end of link.
> 
> Please rethink how you're constructing links here. We shouldn't be replacing the various page URLs. We should instead construct the link directly using the origin, thread id, and chain id (unless on custom domain). The code changes in this PR are extremely hard to follow as a result of the unclear logic.

Basically this PR just needed to add a conditional branch into the onclick for the Copy URL popover menu item for discussion links. Is the unclear logic due to how the onclick is implemented, or is it due to the nature of the PopoverMenu component? Where do you suggest I should I try to refactor?

**Comment by kurtisassad** [comment]
> Also still broken on topics listing pages -- copies of topic at end of link.
> 
> Please rethink how you're constructing links here. We shouldn't be replacing the various page URLs. We should instead construct the link directly using the origin, thread id, and chain id (unless on custom domain). The code changes in this PR are extremely hard to follow as a result of the unclear logic.

I have tried refactoring it a little bit, but I think it will always need to be a bit complicated (or at least have 3 conditional branches). One branch is for comments, one for discussions page, one for thread on a discussion page.

**Comment by kurtisassad** [comment]
I have tried reproducing the bug on custom domains for dydx,osmosis,1inch for threads on discussions page, threads on discussion page, and comment on discussion page for each chain. It seems to work fine for me. Could you let me know where exactly you are running into this bug?

**Comment by jnaviask** [comment]
Overriding forums.dydx.community locally:
- Copying thread URL on discussion page gives https://forums.dydx.community/discussions/discussion/3781-endorsed-delegates-profile-and-pitch
- Copying thread URL on overview page gives https://forums.dydx.community/overview/discussion/10960-test

**Comment by jnaviask** [comment]
@kurtisassad don't we have a helper function for whether we're on a custom domain or not? Can we use that instead?

**Review by jnaviask — APPROVED** [review]
